src/components/NcRichText/NcRichText.vue

```diff
@@ -306,6 +318,21 @@ import rehype2react from 'rehype-react'
 import rehypeExternalLinks from 'rehype-external-links'
 import { RouterLink } from 'vue-router'
 
+/**
+ * Heavy libraries should be loaded on demand to reduce component size
+ */
+let rehypeHighlight // from 'rehype-highlight'
+const rehypeHighlightLoaded = ref(false)
+/**
+ * @param {Function} callback - callback function to execute after library is loaded
+ */
+function importRehypeLibrary(callback) {
+	import('rehype-highlight').then((module) => {
+		rehypeHighlight = module.default
+		callback()
+	})
+}
```

Comment:
You can just return the result and use Promise API, incl. async/await without manually providing callback